var/spack/repos/builtin/packages/ascent/package.py

```diff
@@ -93,27 +93,25 @@ class Ascent(Package, CudaPackage):
     # Certain CMake versions have been found to break for our use cases
     depends_on("cmake@3.14.1:3.14.99,3.18.2:", type='build')
     depends_on("conduit~python", when="~python")
-    depends_on("conduit+python", when="+python+shared")
-    depends_on("conduit~shared~python", when="~shared")
-    depends_on("conduit~python~mpi", when="~python~mpi")
-    depends_on("conduit+python~mpi", when="+python+shared~mpi")
-    depends_on("conduit~shared~python~mpi", when="~shared~mpi")
+    depends_on("conduit+python", when="+python")
+    depends_on("conduit+mpi", when="+mpi")
+    depends_on("conduit~mpi", when="~mpi")
 
     #######################
     # Python
     #######################
     # we need a shared version of python b/c linking with static python lib
     # causes duplicate state issues when running compiled python modules.
-    depends_on("python+shared", when="+python+shared")
-    extends("python", when="+python+shared")
-    depends_on("py-numpy", when="+python+shared", type=('build', 'run'))
-    depends_on("py-pip", when="+python+shared", type=('build', 'run'))
+    depends_on("python+shared", when="+python")
+    extends("python", when="+python")
+    depends_on("py-numpy", when="+python", type=('build', 'run'))
+    depends_on("py-pip", when="+python", type=('build', 'run'))
```

Comment:
Just fyi, this can be written as:
```python
with when('+python'):
    depends_on("python+shared")
    extends('python')
    depends_on('py-numpy', type=('build', 'run'))
    depends_on('py-pip', type=('build', 'run'))
```